python/ray/projects/scripts.py

```diff
@@ -303,23 +336,25 @@ def stop(name):
 def session_start(command, args, shell, name):
     runner = SessionRunner(session_name=name)
     if shell or command:
-        # Get the actual command to run.
-        cmd = runner.format_command(command, args, shell)
+        # Get the actual command to run. This also validates the command,
+        # which should be done before the cluster is started.
+        commands = runner.format_command(command, args, shell)
         num_steps = 4
     else:
         num_steps = 3
 
-    logger.info("[1/{}] Creating cluster".format(num_steps))
-    runner.create_cluster()
-    logger.info("[2/{}] Syncing the project".format(num_steps))
-    runner.sync_files()
-    logger.info("[3/{}] Setting up environment".format(num_steps))
-    runner.setup_environment()
+    for cmd in commands:
+        logger.info("[1/{}] Creating cluster".format(num_steps))
+        runner.create_cluster()
+        logger.info("[2/{}] Syncing the project".format(num_steps))
+        runner.sync_files()
+        logger.info("[3/{}] Setting up environment".format(num_steps))
+        runner.setup_environment()
 
-    if shell or command:
-        # Run the actual command.
-        logger.info("[4/4] Running command")
-        runner.execute_command(cmd)
+        if shell or command:
+            # Run the actual command.
+            logger.info("[4/4] Running command")
+            runner.execute_command(cmd)
```

Comment:
`runner.execute_command` will block until the script finishes. So we are starting multiple session serially right now.

Reply:
Might want to check a non-blocking version? what happen if tmux==True?

Reply:
Yeah, `tmux==True` makes it non-blocking :)

Reply:
I'm now setting this flag if it hasn't already been set and wildcards are used (logging a print statement).